--- tools/pamusb-agent	2013-05-18 16:36:50.664414171 +0200
+++ tools/pamusb-agent.nuevo	2013-05-18 16:39:33.531407641 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (c) 2003-2007 Andrea Luzzardi <scox@sig11.org>
 #
--- tools/pamusb-conf	2013-05-18 16:36:50.664414171 +0200
+++ tools/pamusb-conf.nuevo	2013-05-18 16:39:49.650406995 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (c) 2003-2007 Andrea Luzzardi <scox@sig11.org>
 #

