diff -Npur blender-2.61.orig/build_files/scons/config/linux-config.py blender-2.61/build_files/scons/config/linux-config.py
--- blender-2.61.orig/build_files/scons/config/linux-config.py	2012-01-22 00:59:55.201475928 +0100
+++ blender-2.61/build_files/scons/config/linux-config.py	2012-01-22 01:00:24.211475415 +0100
@@ -189,6 +189,11 @@ BF_EXPAT = '/usr'
 BF_EXPAT_LIB = 'expat'
 BF_EXPAT_LIBPATH = '/usr/lib'
 
+WITH_BF_COLAMD = True
+BF_COLAMD = '#extern/colamd'
+BF_COLAMD_INC = '${BF_COLAMD}/include'
+BF_COLAMD_LIB = 'extern_colamd'
+
 WITH_BF_JEMALLOC = False
 WITH_BF_STATICJEMALLOC = False
 BF_JEMALLOC = '/usr'
diff -Npur blender-2.61.orig/build_files/scons/tools/Blender.py blender-2.61/build_files/scons/tools/Blender.py
--- blender-2.61.orig/build_files/scons/tools/Blender.py	2012-01-22 00:59:55.201475928 +0100
+++ blender-2.61/build_files/scons/tools/Blender.py	2012-01-22 01:00:24.193475419 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+
 
 """
 tools.BlenderEnvironment
@@ -281,6 +281,8 @@ def setup_syslibs(lenv):
         else:
             syslibs += Split(lenv['BF_OPENCOLLADA_LIB'])
         syslibs.append(lenv['BF_EXPAT_LIB'])
+    if lenv['WITH_BF_COLAMD']:
+        syslibs += Split(lenv['BF_COLAMD_LIB'])
 
     if lenv['WITH_BF_JEMALLOC']:
         if not lenv['WITH_BF_STATICJEMALLOC']:
diff -Npur blender-2.61.orig/build_files/scons/tools/btools.py blender-2.61/build_files/scons/tools/btools.py
--- blender-2.61.orig/build_files/scons/tools/btools.py	2012-01-22 00:59:55.201475928 +0100
+++ blender-2.61/build_files/scons/tools/btools.py	2012-01-22 01:01:11.974474571 +0100
@@ -125,6 +125,7 @@ def validate_arguments(args, bc):
             'WITH_BF_FFTW3', 'BF_FFTW3', 'BF_FFTW3_INC', 'BF_FFTW3_LIB', 'BF_FFTW3_LIBPATH', 'WITH_BF_STATICFFTW3', 'BF_FFTW3_LIB_STATIC',
             'WITH_BF_STATICOPENGL', 'BF_OPENGL', 'BF_OPENGL_INC', 'BF_OPENGL_LIB', 'BF_OPENGL_LIBPATH', 'BF_OPENGL_LIB_STATIC',
             'WITH_BF_COLLADA', 'BF_COLLADA', 'BF_COLLADA_INC', 'BF_COLLADA_LIB', 'BF_OPENCOLLADA', 'BF_OPENCOLLADA_INC', 'BF_OPENCOLLADA_LIB', 'BF_OPENCOLLADA_LIBPATH', 'BF_PCRE', 'BF_PCRE_LIB', 'BF_PCRE_LIBPATH', 'BF_EXPAT', 'BF_EXPAT_LIB', 'BF_EXPAT_LIBPATH',
+            'WITH_BF_COLAMD', 'BF_COLAMD', 'BF_COLAMD_INC', 'BF_COLAMD_LIB',
             'WITH_BF_PLAYER',
             'WITH_BF_NOBLENDER',
             'WITH_BF_BINRELOC',
@@ -447,6 +448,11 @@ def read_opts(env, cfg, args):
         ('BF_EXPAT_LIB', 'Expat library', ''),
         ('BF_EXPAT_LIBPATH', 'Expat library path', ''),
         
+        (BoolVariable('WITH_BF_COLAMD', 'Use Colamd if true', True)),
+        ('BF_COLAMD', 'Colamd base dir', ''),
+        ('BF_COLAMD_INC', 'Colamd include path', ''),
+        ('BF_COLAMD_LIB', 'Colamd library', ''),
+
         (BoolVariable('WITH_BF_JEMALLOC', 'Use jemalloc if true', False)),
         (BoolVariable('WITH_BF_STATICJEMALLOC', 'Staticly link to jemalloc', False)),
         ('BF_JEMALLOC', 'jemalloc base path', ''),
diff -Npur blender-2.61.orig/extern/CMakeLists.txt blender-2.61/extern/CMakeLists.txt
--- blender-2.61.orig/extern/CMakeLists.txt	2012-01-22 00:59:55.198475928 +0100
+++ blender-2.61/extern/CMakeLists.txt	2012-01-22 01:00:24.193475419 +0100
@@ -26,8 +26,6 @@
 # Otherwise we get warnings here that we cant fix in external projects
 remove_strict_flags()
 
-add_subdirectory(colamd)
-
 if(WITH_MOD_CLOTH_ELTOPO)
 	add_subdirectory(eltopo)
 endif()
diff -Npur blender-2.61.orig/extern/libmv/bundle.sh blender-2.61/extern/libmv/bundle.sh
--- blender-2.61.orig/extern/libmv/bundle.sh	2012-01-22 00:59:55.194475928 +0100
+++ blender-2.61/extern/libmv/bundle.sh	2012-01-22 01:00:24.194475416 +0100
@@ -118,13 +118,13 @@ set(INC
 	.
 	./third_party/ssba
 	./third_party/ldl/Include
-	../colamd/Include
 )
 
 set(INC_SYS
 	/usr/include/eigen3
 	${PNG_INCLUDE_DIR}
 	${ZLIB_INCLUDE_DIRS}
+	${COLAMD_INCULDE_DIRS}
 )
 
 set(SRC
@@ -227,6 +227,7 @@ $src
 incs = '. /usr/include/eigen3'
 incs += ' ' + env['BF_PNG_INC']
 incs += ' ' + env['BF_ZLIB_INC']
+incs += ' ' + env['BF_COLAMD_INC']
 
 if env['OURPLATFORM'] in ('win32-vc', 'win32-mingw', 'linuxcross', 'win64-vc'):
     incs += ' ./third_party/glog/src/windows ./third_party/glog/src/windows/glog'
@@ -257,7 +258,7 @@ else:
         ccflags_libmv = Split(env['REL_CCFLAGS'])
         cxxflags_libmv = Split(env['REL_CXXFLAGS'])
 
-incs += ' ./third_party/ssba ./third_party/ldl/Include ../colamd/Include'
+incs += ' ./third_party/ssba ./third_party/ldl/Include'
 
 env.BlenderLib ( libname = 'extern_libmv', sources=src, includes=Split(incs), defines=defs, libtype=['extern', 'player'], priority=[20,137], compileflags=cflags_libmv, cc_compileflags=ccflags_libmv, cxx_compileflags=cxxflags_libmv )
 EOF
diff -Npur blender-2.61.orig/extern/libmv/CMakeLists.txt blender-2.61/extern/libmv/CMakeLists.txt
--- blender-2.61.orig/extern/libmv/CMakeLists.txt	2012-01-22 00:59:55.190475928 +0100
+++ blender-2.61/extern/libmv/CMakeLists.txt	2012-01-22 01:00:24.209475415 +0100
@@ -26,13 +26,13 @@ set(INC
 	.
 	third_party/ssba
 	third_party/ldl/Include
-	../colamd/Include
 )
 
 set(INC_SYS
 	/usr/include/eigen3
 	${PNG_INCLUDE_DIR}
 	${ZLIB_INCLUDE_DIRS}
+	${COLAMD_INCLUDE_DIRS}
 )
 
 set(SRC
diff -Npur blender-2.61.orig/extern/libmv/SConscript blender-2.61/extern/libmv/SConscript
--- blender-2.61.orig/extern/libmv/SConscript	2012-01-22 00:59:55.190475928 +0100
+++ blender-2.61/extern/libmv/SConscript	2012-01-22 01:00:24.210475415 +0100
@@ -28,6 +28,7 @@ src += env.Glob('third_party/ssba/Math/*
 incs = '. /usr/include/eigen3'
 incs += ' ' + env['BF_PNG_INC']
 incs += ' ' + env['BF_ZLIB_INC']
+incs += ' ' + env['BF_COLAMD_INC']
 
 if env['OURPLATFORM'] in ('win32-vc', 'win32-mingw', 'linuxcross', 'win64-vc'):
     incs += ' ./third_party/glog/src/windows ./third_party/glog/src/windows/glog'
@@ -57,6 +58,6 @@ else:
         ccflags_libmv += Split(env['REL_CCFLAGS'])
         cxxflags_libmv += Split(env['REL_CXXFLAGS'])
 
-incs += ' ./third_party/ssba ./third_party/ldl/Include ../colamd/Include'
+incs += ' ./third_party/ssba ./third_party/ldl/Include'
 
 env.BlenderLib ( libname = 'extern_libmv', sources=src, includes=Split(incs), defines=defs, libtype=['extern', 'player'], priority=[20,137], compileflags=cflags_libmv, cc_compileflags=ccflags_libmv, cxx_compileflags=cxxflags_libmv )
diff -Npur blender-2.61.orig/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp blender-2.61/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp
--- blender-2.61.orig/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp	2012-01-22 00:59:55.191475928 +0100
+++ blender-2.61/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp	2012-01-22 01:00:24.210475415 +0100
@@ -21,7 +21,7 @@ with SSBA. If not, see <http://www.gnu.o
 
 #if defined(V3DLIB_ENABLE_SUITESPARSE)
 //# include "COLAMD/Include/colamd.h"
-# include "colamd.h"
+# include <colamd.h>
 extern "C"
 {
 //# include "LDL/Include/ldl.h"
diff -Npur blender-2.61.orig/extern/SConscript blender-2.61/extern/SConscript
--- blender-2.61.orig/extern/SConscript	2012-01-22 00:59:55.197475928 +0100
+++ blender-2.61/extern/SConscript	2012-01-22 01:00:24.210475415 +0100
@@ -2,8 +2,6 @@
 
 Import('env')
 
-SConscript(['colamd/SConscript'])
-
 if env['WITH_BF_GAMEENGINE']:
     SConscript(['recastnavigation/SConscript'])
 
diff -Npur blender-2.61.orig/intern/opennl/CMakeLists.txt blender-2.61/intern/opennl/CMakeLists.txt
--- blender-2.61.orig/intern/opennl/CMakeLists.txt	2012-01-22 00:59:55.482475922 +0100
+++ blender-2.61/intern/opennl/CMakeLists.txt	2012-01-22 01:00:24.210475415 +0100
@@ -40,11 +40,10 @@ add_definitions(
 set(INC
 	extern
 	superlu
-	../../extern/colamd/Include
 )
 
 set(INC_SYS
-
+	/usr/include
 )
 
 set(SRC
diff -Npur blender-2.61.orig/intern/opennl/SConscript blender-2.61/intern/opennl/SConscript
--- blender-2.61.orig/intern/opennl/SConscript	2012-01-22 00:59:55.482475922 +0100
+++ blender-2.61/intern/opennl/SConscript	2012-01-22 01:00:24.210475415 +0100
@@ -3,7 +3,7 @@ Import ('env')
 
 sources = env.Glob('intern/*.c') + env.Glob('superlu/*.c')
 
-incs = 'extern superlu ../../extern/colamd/Include'
+incs = 'extern superlu /usr/include'
 
 if (env['OURPLATFORM'] == 'win32-mingw'):
     env.BlenderLib ('bf_intern_opennl', sources, Split(incs), [], libtype=['core','intern'], priority=[1,80] )
diff -Npur blender-2.61.orig/intern/opennl/superlu/get_perm_c.c blender-2.61/intern/opennl/superlu/get_perm_c.c
--- blender-2.61.orig/intern/opennl/superlu/get_perm_c.c	2012-01-22 00:59:55.482475922 +0100
+++ blender-2.61/intern/opennl/superlu/get_perm_c.c	2012-01-22 01:00:24.210475415 +0100
@@ -10,7 +10,7 @@
  */
 
 #include "ssp_defs.h"
-#include "colamd.h"
+#include <colamd.h>
 
 extern int  genmmd_(int *, int *, int *, int *, int *, int *, int *, 
 		    int *, int *, int *, int *, int *);
