--- extern/SConscript-original	2011-12-15 16:25:29.379238136 +0100
+++ extern/SConscript	2011-12-15 16:26:10.376448670 +0100
@@ -2,7 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
 SConscript(['colamd/SConscript'])
 
 if env['WITH_BF_GAMEENGINE']:
--- source/blender/editors/include/BIF_gl.h-original	2011-12-15 16:27:31.299733603 +0100
+++ source/blender/editors/include/BIF_gl.h	2011-12-15 16:30:50.993113607 +0100
@@ -33,7 +33,7 @@
 #ifndef BIF_GL_H
 #define BIF_GL_H
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 
 	/*
 	 * these should be phased out. cpack should be replaced in
--- extern/CMakeLists.txt-original	2011-12-15 16:26:50.104497900 +0100
+++ extern/CMakeLists.txt	2011-12-15 16:28:04.397940215 +0100
@@ -40,9 +40,6 @@
 	add_subdirectory(binreloc)
 endif()
 
-if(WITH_BUILTIN_GLEW)
-	add_subdirectory(glew)
-endif()
 
 if(WITH_GAMEENGINE)
    add_subdirectory(recastnavigation)
--- CMakeLists.txt-original	2011-12-15 16:27:03.309176149 +0100
+++ CMakeLists.txt	2011-12-15 16:29:02.001261302 +0100
@@ -156,7 +156,7 @@
 
 if(UNIX AND NOT APPLE)
 	option(WITH_X11_XINPUT "Enable X11 Xinput (tablet support and unicode input)"	ON)
-	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" ON)
+	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" OFF)
 	option(WITH_XDG_USER_DIRS    "Build with XDG Base Directory Specification (only config and documents for now)" OFF)
 	mark_as_advanced(WITH_XDG_USER_DIRS)
 else()
