--- ono4xx-0.0.3a/ono4xx.c	2011-02-14 20:03:35.000000000 +0100
+++ ono4xx-0.0.4a/ono4xx.c	2011-08-02 20:22:10.000000000 +0200
@@ -219,71 +220,71 @@
           for ( e = 0; e <= 9; ++e )
           {	
 
-			if (a+8>9)	ebpd4 = a-2;
-				else	ebpd4 = a+8;
+		if (a+8>9)	ebpd4 = a-2;
+			else	ebpd4 = a+8;
 
-			ebpd0 = (b*100+a*10+ebp8c+180)/100;		
+		ebpd0 = (b*100+a*10+ebp8c+180)/100;		
 
-			if (c+ebp04>9)	ebpcc = c+ebp04-10;		//REPITE mas abajo para ebpbc
-				else	ebpcc = c+ebp04;
+		if (c+ebp04>9)	ebpcc = c+ebp04-10;		//REPITE mas abajo para ebpbc
+			else	ebpcc = c+ebp04;
 			
-			ebpbc = ebpcc;					//Es igual, calcula lo mismo
+		ebpbc = ebpcc;					//Es igual, calcula lo mismo
 
-			//if (c+ebp04>9)	ebpbc = c+ebp04-10;	// REPITE lo mismo que para ebpcc
-			//	else	ebpbc = c+ebp04;
+		//if (c+ebp04>9)	ebpbc = c+ebp04-10;	// REPITE lo mismo que para ebpcc
+		//	else	ebpbc = c+ebp04;
 
-			ebpc8 = (d*10 + c + ebp08*10 + ebp04)/10;	
+		ebpc8 = (d*10 + c + ebp08*10 + ebp04)/10;	
 
-			if (e*2>9)	ebpc4 = (e-5)*2;
-				else	ebpc4 = e*2;
+		if (e*2>9)	ebpc4 = (e-5)*2;
+			else	ebpc4 = e*2;
 
-			if (d+1>9)	ebpc0 = d-9;			//Nunca se cumple que d+1 sea mayor que 9
-				else	ebpc0 = d+1;			//siempre se usa esto ebpc0 = d+1;
+		if (d+1>9)	ebpc0 = d-9;			//Nunca se cumple que d+1 sea mayor que 9
+			else	ebpc0 = d+1;			//siempre se usa esto ebpc0 = d+1;
 
 
-			aux = b*10+c+ebp04;
+		aux = b*10+c+ebp04;
 			
-			if (aux==-10)	ebpb8 = 0;
-				else	ebpb8 = (aux+10)/10;		
+		if (aux==-10)	ebpb8 = 0;
+			else	ebpb8 = (aux+10)/10;		
 
-			if (a+9>9)	ebpb4 = a-1;
-				else	ebpb4 = a+9;
+		if (a+9>9)	ebpb4 = a-1;
+			else	ebpb4 = a+9;
 			
-			edx = (ebp8c*10+a+9)/10;	//edx
-			if (edx==10)	edx=0;
+		edx = (ebp8c*10+a+9)/10;	//edx
+		if (edx==10)	edx=0;
 
 			
-			if (opcion)
-			{	//------------------------------------------- wpa ------------------------------------------
-				wpa1 = ebp80*100+ebp84*10+ebp88+103;
+		if (opcion)
+		{	//------------------------------------------- wpa ------------------------------------------
+			wpa1 = ebp80*100+ebp84*10+ebp88+103;
 			
-				if (wpa1>1000)	wpa1 = wpa1-897;
+			if (wpa1>1000)	wpa1 = wpa1-897;
 			
-				wpa2 = (ebp8c*100+a*10+b+81)/10;
+			wpa2 = (ebp8c*100+a*10+b+81)/10;
 						
-				fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+			fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
 			
-				if (ebpc0==2)
-					fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0-1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+			if (ebpc0==2)
+				fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0-1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
 
-				if (ebpc0==1)
-					fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0+1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
-			}	//------------------------------------------- wpa ------------------------------------------
-			else 
-			{	//------------------------------------------- wep ------------------------------------------
-				fprintf(fichero,"%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+			if (ebpc0==1)
+				fprintf(fichero,"%i%i%X%X%X%X%X%X%X%X%X\n",wpa1,wpa2,ebpb8,ebpbc,ebpc0+1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+		}	//------------------------------------------- wpa ------------------------------------------
+		else 
+		{	//------------------------------------------- wep ------------------------------------------
+			fprintf(fichero,"%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
 			
-				if (ebpc0==2)
-					fprintf(fichero,"%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0-1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+		   if (ebpc0==2)
+			fprintf(fichero,"%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0-1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);
+
+		if (ebpc0==1)
+			fprintf(fichero,"%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X:%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0+1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);	
+		}	//------------------------------------------- wep ------------------------------------------
 
-				if (ebpc0==1)
-					fprintf(fichero,"%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X%X\n",ebp80,ebp84,ebp88,ebp8c,a,b,c,d,e,ebp08,ebp04,ebp1c,ebp18,ebp14,ebp10,edx,ebpb4,ebpb8,ebpbc,ebpc0+1,ebpc4,ebpc8,ebpcc,ebpd0,ebpd4,ebp8c);	
-			}	//------------------------------------------- wep ------------------------------------------
-
-		  }//for
-		}
-	  }
+	  }//for
 	}
+       }
+     }
   }//for
 			  
 }
