--- share/gpodder/extensions/normalize_audio.py
+++ share/gpodder/extensions/normalize_audio.py
@@ -37,6 +37,9 @@
 }
 
 class gPodderExtension:
+    MIME_TYPES = ('audio/mpeg', 'audio/ogg', )
+    EXT = ('.mp3', '.ogg', )
+
     def __init__(self, container):
         self.container = container
 
@@ -58,13 +61,23 @@ def on_episodes_context_menu(self, episodes):
         if not self.container.config.context_menu:
             return None
 
-        mimetypes = [e.mime_type for e in episodes
-            if e.mime_type is not None and e.file_exists()]
-        if 'audio/ogg' not in mimetypes and 'audio/mpeg' not in mimetypes:
+        if not any(self._check_source(episode) for episode in episodes):
             return None
 
         return [(self.container.metadata.title, self.convert_episodes)]
 
+    def _check_source(self, episode):
+        if not episode.file_exists():
+            return False
+
+        if episode.mime_type in self.MIME_TYPES:
+            return True
+
+        if episode.extension() in self.EXT:
+            return True
+
+        return False
+
     def _convert_episode(self, episode):
         if episode.file_type() != 'audio':
             return
