--- etckeeper.conf	2012-08-25 17:54:21.000000000 +0200
+++ etckeeper.conf	2012-11-10 23:36:05.000000000 +0100
@@ -31,8 +31,10 @@
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum, zypper etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+# For gentoo this is emerge
+HIGHLEVEL_PACKAGE_MANAGER=emerge
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+# For gentoo this is qlist
+LOWLEVEL_PACKAGE_MANAGER=qlist
--- list-installed.d/50list-installed	2012-08-25 17:54:21.000000000 +0200
+++ list-installed.d/50list-installed	2012-11-10 23:37:07.000000000 +0100
@@ -7,4 +7,6 @@
 		egrep '(ok installed|ok config-files)' | cut -f2,3
 elif [ "$LOWLEVEL_PACKAGE_MANAGER" = rpm ]; then
 	rpm -qa --qf "%|epoch?{%{epoch}}:{0}|:%{name}-%{version}-%{release}.%{arch}\n" | sort
+elif [ "$LOWLEVEL_PACKAGE_MANAGER" = qlist ]; then
+	qlist -ICv
 fi
--- Makefile	2012-08-25 17:54:21.000000000 +0200
+++ Makefile	2012-11-10 23:35:17.000000000 +0100
@@ -23,7 +23,6 @@
 	$(INSTALL_DATA) $(CONFFILE) $(DESTDIR)$(etcdir)/etckeeper/etckeeper.conf
 	$(INSTALL_EXE) etckeeper $(DESTDIR)$(bindir)/etckeeper
 	$(INSTALL_DATA) etckeeper.8 $(DESTDIR)$(mandir)/man8/etckeeper.8
-	$(INSTALL_DATA) bash_completion $(DESTDIR)$(etcdir)/bash_completion.d/etckeeper
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),apt)
 	$(INSTALL_DATA) apt.conf $(DESTDIR)$(etcdir)/apt/apt.conf.d/05etckeeper
 	mkdir -p $(DESTDIR)$(etcdir)/cruft/filters-unex
@@ -39,7 +38,6 @@
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),zypper)
 	$(INSTALL_DATA) zypper-etckeeper.py $(DESTDIR)$(prefix)/lib/zypp/plugins/commit/zypper-etckeeper.py
 endif
-	-./etckeeper-bzr/__init__.py install --root=$(DESTDIR) ${PYTHON_INSTALL_OPTS} || echo "** bzr support not installed"
 	echo "** installation successful"
 
 clean: etckeeper.spec
--- update-ignore.d/01update-ignore	2012-08-25 17:54:21.000000000 +0200
+++ update-ignore.d/01update-ignore	2012-11-10 23:38:36.000000000 +0100
@@ -99,8 +99,12 @@
 		ignore "*.pacorig"
 		ignore "*.pacsave"
 		nl
+	elif [ "$LOWLEVEL_PACKAGE_MANAGER" = "qlist" ]; then
+		comment "new and old versions of conffiles, stored by emerge"
+		ignore "*.cfg_*"
+		nl
 	fi
-	
+
 	comment "old versions of files"
 	ignore "*.old"
 	# Not currently ignored as admins tend to rely on these files.
--- post-install.d/50vcs-commit	2012-11-10 22:59:13 +0000
+++ post-install.d/50vcs-commit	2012-11-11 00:32:35 +0000
@@ -1,11 +1,17 @@
 #!/bin/sh
 set -e
-		
+
 pl="/var/cache/etckeeper/packagelist"
 
 if etckeeper unclean; then
+	case $LOWLEVEL_PACKAGE_MANAGER in
+	portage)
+	message="committing changes in /etc after $HIGHLEVEL_PACKAGE_MANAGER run ($CATEGORY/$P)"
+	;;
+	*)
 	message="committing changes in /etc after $HIGHLEVEL_PACKAGE_MANAGER run"
-
+	;;
+	esac
 	if [ -e $pl.pre-install ]; then
 		(
 			echo "$message"
@@ -17,7 +23,7 @@
 		etckeeper commit "$(printf "$message")"
 	fi
 fi
-	
+
 if [ -e $pl.pre-install ]; then
 	rm -f $pl.pre-install
 fi

