--- /var/tmp/portage/x11-misc/xwinwrap-1.1.1-r3/work/xwinwrap/Makefile	2011-08-15 06:20:45.000000000 +0200
+++ xwinwrap/Makefile	2011-08-15 06:18:55.000000000 +0200
@@ -1,11 +1,10 @@
-CFLAGS+=-g -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls
 LDFLAGS+=-lX11 -lXext -lXrender
 
 PROGS=xwinwrap
 
 xwinwrap: xwinwrap.o
 
-all: $(PROGS)
+all: $(PROGS) ${LDFLAGS}
 
 clean:
 	rm -f $(PROGS) *.o
