--- cmake/macros.cmake	2011-11-08 18:33:00.000000000 +0100
+++ cmake/macros.cmake	2011-11-08 18:32:49.000000000 +0100
@@ -93,7 +93,7 @@
     SET(SD_LIBDIR ${_DEFLIBDIR} CACHE PATH "Place where the libraries should go")
     SET(SD_INCLUDEDIR include CACHE PATH "Place where the include files should go")
   ELSE(WIN32) #UNIX
-    SET(SD_BINDIR games CACHE PATH "Place where the executables should go")
+    SET(SD_BINDIR bin CACHE PATH "Place where the executables should go")
     SET(SD_DATADIR share/games/speed-dreams-2 CACHE PATH "Place where all the static data files should go")
     SET(SD_LIBDIR ${_DEFLIBDIR}/games/speed-dreams-2 CACHE PATH "Place where the libraries should go")
     SET(SD_INCLUDEDIR include/speed-dreams-2 CACHE PATH "Place where the include files should go")
