*** scli/vendors.c	2010-02-19 09:27:04.000000000 +0100
--- scli/vendors.c.new	2012-05-06 03:31:14.000000000 +0200
***************
*** 50,55 ****
--- 50,56 ----
      { 2281,	"Ceragon",		"http://www.ceragon.com/" },
      { 2356,	"ELSA GmbH",		"http://www.elsa.de/" },
      { 2636,     "Juniper Networks",     "http://www.juniper.net/" },
+     { 4413,	"Huawei Devices",	"http://www.huawei.com/" },
      { 5567,	"Riverstone Networks",	"http://www.riverstonenet.com/" },
      { 7313,	"1stWave",              "http://www.1stwave.de/" },
      { 8072,	"NET-SNMP",		"http://www.net-snmp.org/" },
